GVFS/GVFS.Service/GvfsService.cs

```diff
@@ -115,18 +119,30 @@ protected override void OnSessionChange(SessionChangeDescription changeDescripti
 
                 if (!GVFSEnlistment.IsUnattended(tracer: null))
                 {
+                    int sessionId = changeDescription.SessionId;
+
                     if (changeDescription.Reason == SessionChangeReason.SessionLogon)
                     {
-                        this.tracer.RelatedInfo("SessionLogon detected, sessionId: {0}", changeDescription.SessionId);
+                        this.tracer.RelatedInfo("SessionLogon detected, sessionId: {0}", sessionId);
                         using (ITracer activity = this.tracer.StartActivity("LogonAutomount", EventLevel.Informational))
                         {
-                            this.repoRegistry.AutoMountRepos(changeDescription.SessionId);
-                            this.repoRegistry.TraceStatus();
+                            var mounter = new RepoAutoMounter(this.tracer, this.repoRegistry, sessionId);
+                            this.repoMounters.Add(sessionId, mounter);
+
+                            mounter.Start();
+
+                            // TODO: this.repoRegistry.TraceStatus();

```

Comment:
Tracing of current state the repositories after logon should probably still happen, but do we want to trace the full status after each 'auto mount' or just at logon?